### PR TITLE
REGEXP_MATCHES and REGEXP_REPLACE function aliases for compatibility with PostgreSQL. 

### LINKED ISSUE #30885: `REGEXP_MATCHES` and `REGEXP_REPLACE` function aliases for compatibility with PostgreSQL.
**Use case**

Some people get used to PostgreSQL. Some tools also.

**Describe the solution you'd like**

Simply add the corresponding aliases. These aliases should be case insensitive.

### PR DESCRIPTION
**Pull request by taiyang-li**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
REGEXP_MATCHES and REGEXP_REPLACE function aliases for compatibility with PostgreSQL.  Close https://github.com/ClickHouse/ClickHouse/issues/30885

### CODE REVIEW

**Comment by kitaisreal** [comment]
https://github.com/Mergifyio update

**Comment by kitaisreal** [comment]
@Mergifyio update

File: src/Functions/match.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@ using FunctionMatch = FunctionsStringSearch<MatchImpl<NameMatch, false>>;
 void registerFunctionMatch(FunctionFactory & factory)
 {
     factory.registerFunction<FunctionMatch>();
+    factory.registerAlias("REGEXP_MATCHES", NameMatch::name);
```
Comment (kitaisreal) [inline src/Functions/match.cpp:23]: We should make alias case insensitive

File: src/Functions/replaceAll.cpp [hunk 2]
```diff
@@ -21,6 +21,7 @@ void registerFunctionReplaceAll(FunctionFactory & factory)
 {
     factory.registerFunction<FunctionReplaceAll>();
     factory.registerAlias("replace", NameReplaceAll::name, FunctionFactory::CaseInsensitive);
+    factory.registerAlias("REGEXP_REPLACE", NameReplaceAll::name);
```
Comment (kitaisreal) [inline src/Functions/replaceAll.cpp:24]: We should make alias case insensitive

**Review by kitaisreal — CHANGES_REQUESTED** [review]
Please add functional test.

File: src/Functions/match.cpp (hunk 1 above)
Comment (taiyang-li) [reply src/Functions/match.cpp:23]: Function name in PostgreSQL is case sensitive, is it better to keep consistent with PostgreSQL in ClickHouse ?

**Comment by taiyang-li** [comment]
> Please add functional test.

Done

**Review by kitaisreal — APPROVED** [review]

File: src/Functions/replaceAll.cpp [hunk 3]
```diff
@@ -21,6 +21,7 @@ void registerFunctionReplaceAll(FunctionFactory & factory)
 {
     factory.registerFunction<FunctionReplaceAll>();
     factory.registerAlias("replace", NameReplaceAll::name, FunctionFactory::CaseInsensitive);
+    factory.registerAlias("REGEXP_REPLACE", NameReplaceAll::name, FunctionFactory::CaseInsensitive);
```
Comment (alexey-milovidov) [inline src/Functions/replaceAll.cpp:24]: This is totally wrong.

File: tests/queries/0_stateless/02205_postgresql_functions.sql [hunk 4]
```diff
@@ -0,0 +1,59 @@
+--- REGEXP_MATCHES
+select match('a key="v" ', 'key="(.*?)"'), REGEXP_MATCHES('a key="v" ', 'key="(.*?)"');
+select match(materialize('a key="v" '), 'key="(.*?)"'), REGEXP_MATCHES(materialize('a key="v" '), 'key="(.*?)"');
+
+select match('\0 key="v" ', 'key="(.*?)"'), REGEXP_MATCHES('\0 key="v" ', 'key="(.*?)"');
+select match(materialize('\0 key="v" '), 'key="(.*?)"'), REGEXP_MATCHES(materialize('\0 key="v" '), 'key="(.*?)"');
+
+
+--- REGEXP_REPLACE
+SELECT s, replaceAll(s, '_', 'o') AS a, REGEXP_REPLACE(s, '_', 'o') AS b, a = b FROM (SELECT arrayJoin(['.', '.']) AS s);
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/02205_postgresql_functions.sql:10]: The tests are wrong as well.

**Comment by alexey-milovidov** [comment]
@kitaisreal and the code review is wrong.

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/ClickHouse/pull/39592

File: src/Functions/match.cpp [hunk 5]
```diff
@@ -20,6 +20,7 @@ using FunctionMatch = FunctionsStringSearch<MatchImpl<NameMatch, false>>;
 void registerFunctionMatch(FunctionFactory & factory)
 {
     factory.registerFunction<FunctionMatch>();
+    factory.registerAlias("REGEXP_MATCHES", NameMatch::name, FunctionFactory::CaseInsensitive);
```
Comment (cpcloud) [inline src/Functions/match.cpp:23]: This seems incorrect. Postgres's `regexp_matches` function doesn't return a boolean indicating whether a pattern matches a string, but returns a set of arrays of matches: https://www.postgresql.org/docs/current/functions-matching.html.

**Pull request merged**